phys/module_sf_bep_bem.F

```diff
@@ -216,15 +215,15 @@ subroutine BEP_BEM(FRC_URB2D,UTYPE_URB2D,itimestep,dz8w,dt,u_phy,v_phy,      &
       integer nzurban(nurbm)
 
 !    Building parameters      
-      real alag_u(nurbm)                      ! Ground thermal diffusivity [m^2 s^-1]
-      real alaw_u(nurbm)                      ! Wall thermal diffusivity [m^2 s^-1]
-      real alar_u(nurbm)                      ! Roof thermal diffusivity [m^2 s^-1]
-      real csg_u(nurbm)                       ! Specific heat of the ground material [J m^3 K^-1]
-      real csw_u(nurbm)                       ! Specific heat of the wall material [J m^3 K^-1]
-      real csr_u(nurbm)                       ! Specific heat of the roof material [J m^3 K^-1]
-      real twini_u(nurbm)                     ! Initial temperature inside the building's wall [K]
-      real trini_u(nurbm)                     ! Initial temperature inside the building's roof [K]
-      real tgini_u(nurbm)                     ! Initial road temperature
+      real,allocatable,dimension(:) :: alag_u                      ! Ground thermal diffusivity [m^2 s^-1]
+      real,allocatable,dimension(:) :: alaw_u                      ! Wall thermal diffusivity [m^2 s^-1]
+      real,allocatable,dimension(:) :: alar_u                      ! Roof thermal diffusivity [m^2 s^-1]
+      real,allocatable,dimension(:) :: csg_u                       ! Specific heat of the ground material [J m^3 K^-1]
+      real,allocatable,dimension(:) :: csw_u                       ! Specific heat of the wall material [J m^3 K^-1]
+      real,allocatable,dimension(:) :: csr_u                       ! Specific heat of the roof material [J m^3 K^-1]
+      real,allocatable,dimension(:) :: twini_u                     ! Initial temperature inside the building's wall [K]
+      real,allocatable,dimension(:) :: trini_u                     ! Initial temperature inside the building's roof [K]
+      real,allocatable,dimension(:) :: tgini_u                     ! Initial road temperature
 
```

Comment:
@andreazonato  To me, this change here along with similar changes to variable declaration is not necessary, since the value of nurbm is already passed to the BEP and BEP_BEM subroutine. Am I right?

Reply:
Yes, it would be possible to use the use_wudapt_lcz flag instead. But avoiding to using it, I think it is faster since we don't have to pass the variable from a routine to an other.